### DISCUSSION TITLE
How to hide widgets without having them take space on the page?

### DISCUSSION THREAD

**Discussion by ryanjackgit** [slint-ui/slint#3354] [Q&A]
--

**Comment by ogoffart** [comment]
(I converted the issue into a discussion, because Issue are meant for bug tracking, and question should be filled in discussions)

You can use the conditional element (`if ... :`)
https://slint.dev/releases/1.1.1/docs/slint/src/reference/conditions 


```
// This button won't be visible or taking space on the page
if (false) : Button { ... }
```

**Comment by ryanjackgit** [comment]
I have a login form, It's two text editor,username and password. After user submit this form,if username and password is right,This login form will be disappear. The main window shows now.  How to implemention it? use popwindow? but popwindow is not modal dailog.
Thanks.

**Reply by ogoffart** [reply]
PopupWindow is kind of a model dialog.
There is an example here in the `todo` example in this repository, when you close the window it shows a popup to ask you if you want to save.

But there are many other ways to do it:  having a pane on top, or two windows.  https://github.com/slint-ui/slint/pull/2094 shows a way to have multiple windows

**Reply by ryanjackgit** [reply]
I can do it now, I use two windows,One is login window,The other is main window.
Thanks a lot.
